### PR TITLE
Remove `jquery.inview` JavaScript library

### PR DESCRIPTION
**Pull request by adamzap**
The `jquery.inview` library was being used to trigger an animation when an icon in a feature list element is fully visible on screen. This patch replaces `jquery.inview` with an implementation based on `IntersectionObserver`, which is widely supported: https://caniuse.com/intersectionobserver.

This patch also simplifies the structure of `list-feature.js`.

### CODE REVIEW

File: djangoproject/static/js/mod/list-feature.js
Comment (adamzap) [inline djangoproject/static/js/mod/list-feature.js:1]: Let me know if anything should change here style-wise.

**Comment by adamzap** [comment]
To verify that this works, you'll want to check that the icons on the homepage and the overview page animate as before:

image

image

File: djangoproject/static/js/mod/list-feature.js [hunk 1]
```diff
@@ -1,23 +1,18 @@
...
-                }
-            });
-        }
-    };
+            observer.unobserve(entry.target);
+        });
 
-    // Export a single instance of our module:
-    return new FeatureList('.list-features');
+    }, { threshold: 1.0 });
```
Comment (bmispelon) [inline djangoproject/static/js/mod/list-feature.js:13]: A comment similar to the previous `// element completely visible` would be welcome here (had to look up the specs to understand what `threshold` meant)

File: djangoproject/static/js/mod/list-feature.js [hunk 2]
```diff
@@ -1,23 +1,18 @@
...
+
+    $('.list-features').find('i').each(function () {
```
Comment (bmispelon) [inline djangoproject/static/js/mod/list-feature.js:15]: Isn't this equivalent to:
```suggestion
    $('.list-features i').each(function () {
```
?

File: djangoproject/static/js/mod/list-feature.js
Comment (bmispelon) [reply djangoproject/static/js/mod/list-feature.js:1]: No comment on the style, other than I think we should use an autoformatter so we don't have to think about it (I thought the project had `prettier` configured in pre-commit, but maybe it's not run on js files? 🤔 )

That's out of scope for this PR though.

**Review by bmispelon — COMMENTED** [review]
Another really nice cleanup 🚀 

It's pretty amazing to see how far the web has come in the 10 years since this design was made. All these jquery plugins are now either not needed or can be replaced by native APIs.

I just had two very minor comments. The functionality is indeed working. 👍🏻

File: djangoproject/static/js/mod/list-feature.js (hunk 2 above)
Comment (adamzap) [reply djangoproject/static/js/mod/list-feature.js:15]: Oh yes, thanks!

Comment (adamzap) [reply djangoproject/static/js/mod/list-feature.js]: Fixed in b24f10ba.

File: djangoproject/static/js/mod/list-feature.js (hunk 1 above)
Comment (adamzap) [reply djangoproject/static/js/mod/list-feature.js]: Fixed in 7b85d55d.

**Review by bmispelon — APPROVED** [review]
Excellent 👍🏻

**Pull request merged**